### PR TITLE
add init_number initialization for dyups

### PR DESCRIPTION
**Pull request by FengXingYuXin**
after updating upstream by dyups restful api, the servers' weight incline

### CODE REVIEW

**Comment by wangfakang** [comment]
@FengXingYuXin Please resolve the conflict because we recently updated the core code. 
THX

**Comment by FengXingYuXin** [comment]
@wangfakang Thanks for your remind and  I have resolved it.

**Comment by wangfakang** [comment]
All tests successful.     ACK

```
$ TEST_NGINX_BINARY=/home/fakang.wfk/work/github/tengine/objs/nginx  prove -v -I tests/nginx-tests/nginx-tests/lib/  ./tests/nginx-tests/cases/dyups.t
./tests/nginx-tests/cases/dyups.t .. 
1..78
your test dir is /tmp/nginx-test-J3VLSL0QuT at ./tests/nginx-tests/cases/dyups.t line 85.
ok 1 - 2013-02-26 16:51:46
ok 2 - 2013-02-26 17:30:07
ok 3 - 2013-02-26 17:35:19
ok 4 - 2013-02-26 16:51:51
ok 5 - 2013-02-26 16:51:54
ok 6 - 2013-02-26 17:36:42
ok 7 - 2013-02-26 17:36:46
ok 8 - 2013-02-26 17:02:13
ok 9 - 2013-02-26 17:36:59
ok 10 - 2013-02-26 17:40:24
ok 11 - 2013-02-26 17:40:28
ok 12 - 2013-02-26 17:40:32
ok 13 - 2013-02-26 17:40:36
ok 14 - 2013-02-26 17:40:39
ok 15 - 2013-02-26 17:41:09
ok 16 - 2013-02-26 16:51:57
ok 17 - 2013-02-26 16:52:00
ok 18 - 2013-02-26 17:00:54
ok 19 - 2013-02-26 17:42:27
ok 20 - 2013-02-26 17:44:34
ok 21 - 2013-02-26 17:08:00
ok 22 - 2013-02-26 17:08:04
ok 23 - 2013-02-26 17:45:03
ok 24 - 2013-02-26 17:05:20
ok 25 - 2013-02-26 17:05:30
ok 26 - 2013-06-20 17:46:03
ok 27 - 2013-02-28 16:27:45
ok 28 - 2013-02-28 16:27:49
ok 29 - 2013-02-28 16:27:52
ok 30 - 2013-02-28 16:28:44
ok 31 - 2013-02-28 16:23:41
ok 32 - 2013-02-28 16:23:48
ok 33 - 2013-02-28 16:25:32
ok 34 - 2013-02-28 16:25:35
ok 35 - 2013-03-04 15:53:41
ok 36 - 2013-03-04 15:53:44
ok 37 - 2013-03-04 15:53:46
ok 38 - 2013-03-04 15:53:49
ok 39 - 2013-03-05 15:36:40
ok 40 - 2013-03-05 15:37:25
ok 41 - 2013-03-05 16:13:11
ok 42 - 2013-03-05 16:13:23
ok 43 - 2013-03-25 10:29:48
ok 44 - 2013-03-25 10:39:02
ok 45 - 2013-03-25 10:39:28
ok 46 - 2013-03-25 10:49:44
ok 47 - 2013-03-25 10:49:47
ok 48 - 2013-03-25 10:50:41
ok 49 - 2013-03-25 10:49:51
ok 50 - 2014-06-15 07:45:30
ok 51 - 2014-06-15 07:45:33
ok 52 - 2014-06-15 07:45:40
ok 53 - 2014-06-15 07:45:43
nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
ok 54 - 2013-02-26 16:51:46
ok 55 - 2013-02-26 17:30:07
ok 56 - 2013-02-26 17:35:19
ok 57 - 2013-02-26 16:51:51
ok 58 - 2013-02-26 17:02:13
ok 59 - 2013-02-26 17:36:59
ok 60 - 2013-02-26 17:40:24
ok 61 - 2013-02-26 17:41:09
ok 62 - 2013-02-26 17:00:54
ok 63 - 2013-02-26 17:42:27
ok 64 - 2013-02-26 17:45:03
ok 65 - 2013-02-26 17:05:20
ok 66 - 2013-02-26 17:46:03
ok 67 - 2013-02-28 16:27:45
ok 68 - 2013-02-28 16:23:41
ok 69 - 2013-03-04 15:53:41
ok 70 - 2013-03-05 15:36:40
ok 71 - 2013-03-05 15:37:25
ok 72 - 2013-03-05 16:13:11
nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
ok 73 - 5/ 5 11:04:49 2014
ok 74 - 5/ 5 11:04:42 2014
ok 75 - 5/ 5 11:08:08 2014
ok 76 - 5/ 5 11:08:16 2014
ok 77 - no alerts
ok 78 - no sanitizer errors
ok
All tests successful.
Files=1, Tests=78,  8 wallclock secs ( 0.05 usr  0.02 sys +  0.22 cusr  0.12 csys =  0.41 CPU)
Result: PASS
```

**Comment by chobits** [comment]
ACK

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 1]
```diff
@@ -1426,6 +1427,14 @@ ngx_dyups_add_server(ngx_http_dyups_srv_conf_t *duscf, ngx_buf_t *buf)
...
         return NGX_ERROR;
     }
+   
+    /*add init_number initialization*/
+    peers = uscf->peer.data;
+    peers->init_number = ngx_random() % peers->number;
+    backup = peers->next;
+    if (backup) {
+        backup->init_number = ngx_random() % backup->number;
+    }
```
[the remark this answers is not in the document]
Comment (chobits) [reply modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: fixed in https://github.com/alibaba/tengine/pull/1247.

**Pull request merged**